src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/update/actions/RemoveOldRulesAction.java

```diff
@@ -57,27 +57,35 @@ protected void perform(State from, State to, Event event, FlowUpdateContext cont
 
         Collection<FlowSegmentRequestFactory> factories = new ArrayList<>();
 
-        Flow flow = RequestedFlowMapper.INSTANCE.toFlow(stateMachine.getOriginalFlow());
```

Comment:
copy & paste is evil - you can move this chunk in parent class.

Reply:
Reworked.